File: problems/problems_1784/solution.go
package problem1784

import (
	"encoding/json"
	"log"
	"strings"
)

func checkOnesSegment(s string) bool {
	count := 0
	appear := false
	for _, r := range s {
		if r == '1' {
			appear = true
		} else {
			if appear {
				count++
				if count > 1 {
					return false
				}
				appear = false
			}
		}
	}
	return count == 0 || !appear
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var s string

	if err := json.Unmarshal([]byte(inputValues[0]), &s); err != nil {
		log.Fatal(err)
	}

	return checkOnesSegment(s)
}

File: problems/problems_1784/Solution.cpp
//go:build ignore
#include "cpp/common/Solution.h"

using namespace std;
using json = nlohmann::json;

class Solution {
public:
  bool checkOnesSegment(const string& s) {
    int count = 0;
    bool appear = false;
    for (char c : s) {
      if (c == '1') {
        appear = true;
      } else {
        if (appear) {
          if (++count > 1) {
            return false;
          }
          appear = false;
        }
      }
    }
    return count == 0 || !appear;
  }
};

json leetcode::qubh::Solve(string input_json_values) {
  vector<string> inputArray;
  size_t pos = input_json_values.find('\n');
  while (pos != string::npos) {
    inputArray.push_back(input_json_values.substr(0, pos));
    input_json_values = input_json_values.substr(pos + 1);
    pos = input_json_values.find('\n');
  }
  inputArray.push_back(input_json_values);

  Solution solution;
  string s = json::parse(inputArray.at(0));
  return solution.checkOnesSegment(s);
}
